### PR TITLE
[TensorRT] query GPU properties only once when setting device_id

### PR DESCRIPTION
**Pull request by gedoensmax**
### Description

For most models this does not show significant overhead but for very small models it shows significant impact. 
Attached screenshot shows impact when only using 2 FC layers: 
image

### CODE REVIEW

**Comment by gedoensmax** [comment]
@chilo-ms Could you review this ? I hope it does not collide with the other PR otherwise let me rebase this one or you cherry pick my changes :)

**Review by tianleiwu — APPROVED** [review]

**Review by jywu-msft — APPROVED** [review]

**Comment by jywu-msft** [comment]
This will conflict with @chilo-ms 's pending change. but i think this change is relatively straightforward.
will let him decide whether to merge this first or after his gets merged in.

**Pull request merged**
